RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,111 @@
+   there is a missing label, please add one.
+1. Choose an emoji for the release. Try to make it semi-related to something that's been included in the release (point releases can be a little weirder).
+1. Add this release to the `CHANGELOG.md`. Use the structure of previous
+   entries. If you use VS Code, you can use [this snippet](https://gist.github.com/EverlastingBugstopper/04d1adb99506388ff9d7abd8d0a82bc3) to insert new changelog sections. If it is a release candidate, no official changelog is needed, but testing instructions will be added later in the process.
+
+### Start a release PR
+
+1. Create a new branch "#.#.#" where "#.#.#" is this release's version (release) or "#.#.#-rc.#" (release candidate)
+1. Push up a commit with the `CHANGELOG.md` changes. The commit message can just be "#.#.#" (release) or "#.#.#-rc.#" (release candidate)
+1. Request review from the @cloudflare/workers-devexp team.
+
+### Review
+
+Most of your comments will be about the changelog. Once the PR is finalized and approved...
+
+1. If you made changes, squash or fixup all changes into a single commit.
+1. Run `git push` and wait for CI to pass.
+
+### Tag and build release
+
+This part of the release process is handled by GitHub Actions, and our binaries are distributed as GitHub Releases. When you push a version tag, it kicks off an action that creates a new GitHub release for that tag, builds release binaries and attaches them to the release.
+
+1. Once ready to merge, tag the commit by running either `git tag -a v#.#.# -m #.#.#` (release), or `git tag -a v#.#.#-rc.# -m #.#.#` (release candidate)
+1. Run `git push --tags`.
+1. Wait for CI to pass.
+
+### Edit the release
+
+After CI builds the release binaries and they appear on the [releases page](https://github.com/cloudflare/kv-asset-handler/releases), click `Edit` and update release notes.
+
+### Publish to npm
+
+Full releases are tagged `latest`. Release candidates are tagged `beta`. If for some reason you mix up the commands below, follow the troubleshooting guide.
+
+1. If this is a full release, `cd npm && npm publish`. If it is a release candidate, `cd npm && npm publish --tag beta`
+1. Tweet.
+
+# Troubleshooting a release
+
+Mistakes happen. Most of these release steps are recoverable if you mess up. The goal is not to, but if you find yourself cursing a fat fingered command, here are some troubleshooting tips. Please feel free to add to this guide.
+
+## I pushed the wrong tag
+
+Tags and releases can be removed in GitHub. First, [remove the remote tag](https://stackoverflow.com/questions/5480258/how-to-delete-a-remote-tag):
+
+```console
+$ git push --delete origin tagname
+```
+
+This will turn the release into a `draft` and you can delete it from the edit page.
+
+Make sure you also delete the local tag:
+
+```console
+$ git tag --delete vX.X.X
+```
+
+## I forgot to add the `beta` tag to my RC when I ran `npm publish`
```

Comment:
this is specific to RCs so can probably be dropped. a section around un-publishing and/or deprecating may be useful here, maybe even in both repos.